cheroot/connections.py

```diff
@@ -133,7 +132,7 @@ def get_conn(self, server_socket):
 
         # We have a connection ready for use.
         if conn:
-            del self.connections[conn]
+            self.connections.remove(conn)
```

Comment:
Removal of an item from a list in Python is expensive in general. Especially, from its beginning: all following elements are shifted in memory. So OrderedDict was a more effective solution.

Reply:
> Weren't you only manipulating it from a single thread?
No. The main thread was cycling through open connections, but worker threads were also putting back non-closed connections.